src/NuGet.CommandLine.StagingExtensions/StageCommand.cs

```diff
@@ -146,6 +179,144 @@ private bool TryParseArguments(out SubCommand subCommand, out string parameter)
             }
         }
 
+        private async Task GetStageDetails(StageManagementResource stageManagementResource, string stageId)
+        {
+            var stageDetails = await stageManagementResource.Get(stageId, Console);
+
+            Console.WriteLine();
+            Console.PrintJustified(0, string.Format(CultureInfo.CurrentCulture, "{0}: {1} ({2})", StagingResources.HeaderStageName, stageDetails.DisplayName, stageDetails.Id));
+            Console.PrintJustified(1, string.Format(CultureInfo.CurrentCulture, "{0}: {1}", StagingResources.HeaderStageStatus, stageDetails.Status));
+            Console.PrintJustified(1, string.Format(CultureInfo.CurrentCulture, "{0}: {1}", StagingResources.HeaderStageFeed, stageDetails.Feed));
+            Console.PrintJustified(1, string.Format(CultureInfo.CurrentCulture, "{0}: {1}", StagingResources.HeaderStageCreationDate, stageDetails.CreationDate));
+            Console.PrintJustified(1, string.Format(CultureInfo.CurrentCulture, "{0}: {1}", StagingResources.HeaderStageExpirationDate, stageDetails.ExpirationDate));
+            Console.PrintJustified(1, string.Format(CultureInfo.CurrentCulture, "{0}:", StagingResources.HeaderStageMemberships));
+
+            foreach (var membership in stageDetails.Memberships)
+            {
+                Console.PrintJustified(2, string.Format(CultureInfo.CurrentCulture, "{0}: {1}", StagingResources.HeaderStageMemberUserName, membership.Name));
+                Console.PrintJustified(2, string.Format(CultureInfo.CurrentCulture, "{0}: {1}", StagingResources.HeaderStageMemberType, membership.MembershipType));
+            }
+
+            Console.PrintJustified(1, string.Format(CultureInfo.CurrentCulture, "{0}:", StagingResources.HeaderStagePackages));
+
+            if (stageDetails.PackagesCount > 0)
+            {
+                foreach (var package in stageDetails.Packages)
+                {
+                    Console.PrintJustified(2, string.Format(CultureInfo.CurrentCulture, "{0}, {1}", package.Id, package.Version));
+                }
+            }
+            else
+            {
+                Console.PrintJustified(2, string.Format(CultureInfo.CurrentCulture, "{0}", StagingResources.None));
+            }
+        }
+
+        private async Task GetCommitProgress(StageManagementResource stageManagementResource, string stageId)
+        {
+            const int checkIntervalSeconds = 10;
+            bool done = false;
+
+            while (!done)
+            {
+                var progress = await stageManagementResource.GetCommitProgress(stageId, Console);
+
+                if (progress.CommitStatus == CommitPending)

```

Comment:
I feel `Switch` statement would be cleaner. @scottbommarito - I think we can use strings as well in switch statements.